### PR TITLE
Sort peers to download blocks by height

### PR DESCRIPTION
**Pull request by earlbread**

### CODE REVIEW

**Comment by moreal** [comment]
I also considered you did. But I didn't apply it and stay with the ordering from *Kademlia* because of possibility preloading can be concentrated on one node having the most blocks. How do you think about that?

File: Libplanet/Net/Swarm.cs [hunk 1]
```diff
@@ -693,6 +693,8 @@ await DialToExistingPeers(dialTimeout, cancellationToken)
                 return;
             }
 
+            peersWithHeight = peersWithHeight.OrderByDescending(p => p.Item2).ToList();
```
Comment (dahlia) [inline Libplanet/Net/Swarm.cs]: How about sorting it before it's become realized (i.e., before `.ToListAsync()` being called) rather than here?

**Comment by earlbread** [comment]
> I also considered you did and it will help your chain keep up with the chain of network. But I didn't apply it and stay with the ordering from _Kademlia_ because of possibility preloading can be concentrated on one node having the most blocks. How do you think about that?

Makes sense. But in general, I think many peers have blocks of the same height. Don't they?

File: Libplanet/Net/Swarm.cs (hunk 1 above)
Comment (earlbread) [reply Libplanet/Net/Swarm.cs]: I tried to but couldn't. Maybe because `peersWithHeight` is `IAsyncEnumerable`.

**Comment by moreal** [comment]
> > I also considered you did and it will help your chain keep up with the chain of network. But I didn't apply it and stay with the ordering from _Kademlia_ because of possibility preloading can be concentrated on one node having the most blocks. How do you think about that?
> 
> Makes sense. But in general, I think many peers have blocks of the same height. Don't they?

Right.

**Pull request merged**